system/views/general/_adminToolbar.cfm

```diff
@@ -94,6 +94,9 @@
 					</cfif>
 					<div class="navbar-header pull-right">
 						<ul class="nav ace-nav">
+							<cfif #event.isImpersonated()#>
+								<li>&nbsp; <label class="text-success">Impersonating</label> : #getLoggedInUserDetails().email_address#&nbsp;</li>
```

Comment:
This should be `i18n`:

```cfm
<li>&nbsp; <i class="fa fa-fw fa-user-md green"></i> #translateResource( uri="cms:admin.bar.impersonating.web.user", data=[ getLoggedInUserDetails().display_name ]  )#</li>
```

Reply:
Hi @DominicWatson ,
Could we use email address in order to maintain the uniqueness.